tests/test_downloader_handlers.py

```diff
@@ -45,20 +44,32 @@ class DummyDH(object):
     def __init__(self, crawler):
         pass
 
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler)
+
 
 class DummyLazyDH(object):
     # Default is lazy for backward compatibility
 
     def __init__(self, crawler):
         pass
 
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler)
+
 
 class OffDH(object):
     lazy = False
 
     def __init__(self, crawler):
         raise NotConfigured
 
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler)
```

Comment:
Are these methods used? Can you remove from_crawler and `__init__` methods from all DHs, and heep `__init__` with an exception only here?